### PR TITLE
8334397: RISC-V: verify perf of ReverseBytesS/US

### PR DESCRIPTION
**Pull request by Hamlin-Li**
Hi,
Can you help to review this patch?
Thanks!

This is similar with previous JDK-8334396.
Added some tests.

### Test
<google-sheets-html-origin style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">
  | Tests | Scores | Errors | Unit
-- | -- | -- | -- | --
Intrinsic, +zbb, +rvv | Characters.reverseBytes | 1654.535 | 69.36 | ns/op
  | Shorts.reverseBytes | 1795.403 | 44.015 | ns/op
Intrinsic, +zbb, -rvv | Characters.reverseBytes | 1649.752 | 74.965 | ns/op
  | Shorts.reverseBytes | 1798.637 | 49.52 | ns/op
Intrinsic, -zbb, +rvv | Characters.reverseBytes | 2279.588 | 44.222 | ns/op
  | Shorts.reverseBytes | 2441.674 | 63.895 | ns/op
Intrinsic, -zbb, -rvv | Characters.reverseBytes | 2288.876 | 49.099 | ns/op
  | Shorts.reverseBytes | 2454.454 | 94.004 | ns/op
No intrinsic | Characters.reverseBytes | 1629.722 | 23.656 | ns/op
  | Shorts.reverseBytes | 2108.81 | 43.378 | ns/op

</google-sheets-html-origin>

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8334397](https://bugs.openjdk.org/browse/JDK-8334397): RISC-V: verify perf of ReverseBytesS/US (**Sub-task** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Ludovic Henry](https://openjdk.org/census#luhenry) (@luhenry - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19830/head:pull/19830` \
`$ git checkout pull/19830`

Update a local copy of the PR: \
`$ git checkout pull/19830` \
`$ git pull https://git.openjdk.org/jdk.git pull/19830/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19830`

View PR using the GUI difftool: \
`$ git pr show -t 19830`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19830.diff">https://git.openjdk.org/jdk/pull/19830.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19830#issuecomment-2182866211)

### CODE REVIEW

**Review by RealFYang — APPROVED** [review]
LGTM. Thanks.

File: test/micro/org/openjdk/bench/java/lang/Characters.java [hunk 1]
```diff
@@ -48,6 +49,29 @@ public class Characters {
...
+        Random r  = new Random(0);
+        chars     = new char[size];
+        res       = new char[size];
+        for (int i = 0; i < size; i++) {
+            chars[i] = (char)r.nextInt(Character.MAX_VALUE + 1);
+        }
+    }
+
+    @Benchmark
+    public void reverseBytes() {
```
Comment (luhenry) [inline test/micro/org/openjdk/bench/java/lang/Characters.java:69]: I'm not sure we want to be adding that benchmark to this class. Could you move to a different class that will test exclusively `reverseBytes` on `char[]`? You can then move the code from `test/micro/org/openjdk/bench/java/lang/Shorts.java` into that same class or a similarly named class for `short[]`.

Comment (Hamlin-Li) [reply test/micro/org/openjdk/bench/java/lang/Characters.java:69]: Not sure if I understand you correctly.
The test is for reverseBytes of a char, not char[]. And it's quite similar as existing test in for integer at: https://github.com/openjdk/jdk/blob/master/test/micro/org/openjdk/bench/java/lang/Integers.java#L171

Comment (luhenry) [reply test/micro/org/openjdk/bench/java/lang/Characters.java:69]: I meant put this `reverseBytes` benchmark in a different class, but given it's how it's done in https://github.com/openjdk/jdk/blob/master/test/micro/org/openjdk/bench/java/lang/Integers.java#L171, please ignore this comment.

**Review by luhenry — APPROVED** [review]

**Comment by Hamlin-Li** [comment]
Thanks @luhenry @RealFYang for your reviewing.

**Pull request closed without merging**